test/addons-napi/test_async/test.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+const test_array = require(`./build/${common.buildType}/test_array`);
+
+test_array(5, common.mustCall(function(err, val) {
+  assert.strictEqual(err, null);
+  assert.strictEqual(val, 10);
+  process.nextTick(common.mustCall(function() {}));
```

Comment:
This line is a bit odd… does it serve some particular purpose? It brings up a good point – ideally, the C callback should use `napi_make_callback` instead of `napi_call_function`…

It might even be better if the `complete` callback would always be called in a MakeCallback-ish way, but that probably requires some refactoring of `MakeCallback` itself.